src/test/java/RobotWarSimulationTest.java

```diff
@@ -0,0 +1,59 @@
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import skynet.Faction;
+import skynet.Factory;
+import skynet.Part;
+
+import java.util.Random;
+import static org.junit.jupiter.api.Assertions.*;
+
+public class RobotWarSimulationTest {
+    private Factory factory;
+    private Faction world;
+    private Faction wednesday;
+
+    @BeforeEach
+    void setUp() {
+        factory = new Factory();
+        world = new Faction("World", factory, 5);
+        wednesday = new Faction("Wednesday", factory, 5);
+    }
+
+    @Test
+    void testFactoryProducesParts() {
+        Random fixedRandom = new Random(0);
+        factory.produceParts(fixedRandom);
+        assertFalse(factory.getParts().isEmpty());
+        assertTrue(factory.getParts().size() <= 10);
+    }
+
+    @Test
+    void testFactionTakesUpToFiveParts() {
+        for (int i = 0; i < 20; i++) {
```

Comment:
The test is directly modifying the factory's parts list through factory.getParts().add() which bypasses the factory's production logic and breaks encapsulation.